openerp/addons/base/res/res_partner.py

```diff
@@ -387,6 +393,18 @@ def onchange_state(self, state_id):
             return {'value': {'country_id': state.country_id.id}}
         return {'value': {}}
 
+    @api.onchange('image')
+    def onchange_email(self):
+        if not self.image and self.email:
```

Comment:
So the gravatar is not updated when the email is changed? Why not just use the gravatar as default value then?